### PR TITLE
Project 16 - Darcy & Rasmus

### PR DESCRIPTION
**Pull request by darcywinterhill**
Link to my deployed project:

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/components/GameOverScreen.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        <GameOverTitle>
+          GAME OVER
+        </GameOverTitle>
+        <GameOverText>
+          You went in circles and came back to where you started.
+          Better luck next time...
+        </GameOverText>
+        <Icon className='nes-icon close is-large'>
+        </Icon>
+        <Link to='/'>
```
Comment (Ann-Sofi) [inline code/src/components/GameOverScreen.js:21]: Nice that you used router!

File: code/src/pages/GameScreen.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  return (
+    <>
+      {isLoading &&
+        <LoadingScreen />
+      }
+      {!isLoading && (
+        <>
+          {error &&
+            `Ooops, something went wrong: ${error}`
```
Comment (Ann-Sofi) [inline code/src/pages/GameScreen.js:23]: Good that you handle errors!

File: code/src/pages/GameScreen.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  const isLoading = useSelector((store) => store.maze.isLoading)
+  const error = useSelector((store) => store.maze.error)
+  const gameState = useSelector((store) => store.maze.gameState)
+
+  return (
+    <>
+      {isLoading &&
+        <LoadingScreen />
+      }
+      {!isLoading && (
```
Comment (Ann-Sofi) [inline code/src/pages/GameScreen.js:20]: One other way of doing it could be to use ternery operator. :-)

File: code/src/pages/StartScreen.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  const onStartGame = () => {
+      dispatch(maze.actions.setLoading(!isLoading))
+      dispatch(maze.actions.setUsername(inputValue))
+      dispatch(firstFetch());
+  }
+  
+  return (
+    <Main>
+      <Border>
+        <Greeting>
```
Comment (Ann-Sofi) [inline code/src/pages/StartScreen.js:25]: Good descriptive naming of styled components!

File: code/src/reducers/maze.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+      const setInitialState = ''
+      store.username = setInitialState
+      store.gameState = setInitialState
+    }
+  }
+})
+
+export default maze
+
+export const firstFetch = () => {
```
Comment (Ann-Sofi) [inline code/src/reducers/maze.js:52]: Namingthe thunks first and second fetch is not very descriptive of what they do. ;-)

**Review by Ann-Sofi — APPROVED** [review]
Really great job!! Nice structure and good use of both components and styled components.  
Very cool retro style and we especially like the hearts!

A small comment about the styling: The direction-containers jumps around on the screen depending on how big the bubble is. 
And the loading symbol is only visable after pressing start and not between the different rooms.

File: code/src/pages/StartScreen.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+          Enter your name and start the adventure.
+        </Instruction>
+        <Label
+          htmlFor='name_input'>
+        <NameInput
+          type='text'
+          id='name_input'
+          value={inputValue}
+          onChange={event => setInputValue(event.target.value)}
+        />
```
Comment (KarolCislowski) [inline code/src/pages/StartScreen.js:38]: It is not nice that you forgot about the form tag. It striped your input from submission on enter, what is the most used sending the form method 😭

**Review by KarolCislowski — APPROVED** [review]
Great app 🥳

**Pull request closed without merging**